storey/aggregations.py

```diff
@@ -240,17 +240,20 @@ def __init__(self, key, aggregates, base_time, initial_data=None):
 
         # Add all raw aggregates, including aggregates not explicitly requested.
         for aggregation_metadata in aggregates:
-            for aggr, is_hidden in get_all_raw_aggregates_with_hidden(aggregation_metadata.aggregations).items():
-                column_name = f'{aggregation_metadata.name}_{aggr}'
-                if column_name in self.aggregation_buckets:
-                    self.aggregation_buckets[column_name].is_hidden &= is_hidden
-                else:
-                    initial_column_data = None
-                    if initial_data and column_name in initial_data:
-                        initial_column_data = initial_data[column_name]
-                    self.aggregation_buckets[column_name] = \
-                        AggregationBuckets(aggregation_metadata.name, aggr, aggregation_metadata.windows, base_time,
-                                           aggregation_metadata.max_value, is_hidden, initial_column_data)
+            for meta in aggregation_metadata.aggregations:
+                for aggr, is_hidden in get_all_raw_aggregates_with_hidden([meta]).items():
+                    column_name = f'{aggregation_metadata.name}_{aggr}'
+                    if column_name in self.aggregation_buckets:
+                        self.aggregation_buckets[column_name].add_to_window(is_hidden, copy.deepcopy(aggregation_metadata.windows))
```

Comment:
i think it's better to go over the aggregations once and "aggregate" all the windows and only then create the aggregation buckets with the already finite windows. the `add_to_window` logic shouldn't really be a part of the AggregationBuckets, it's not updating during the flow just once when it's initialized